dev/ComboBox/ComboBoxHelper.cpp

```diff
@@ -0,0 +1,159 @@
+// Creating this attached property to help us plug in some extra logic without touching the actual ComboBox code.
+void ComboBoxHelper::OnApplyDynamicCornerRadiusPropertyChanged(
+    const winrt::DependencyObject& sender,
+    const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (auto comboBox = sender.try_as<winrt::ComboBox>())
+    {
+        if (auto revokersInspectable = comboBox.GetValue(DropDownEventRevokersProperty()))
+        {
+            if (auto revokers = revokersInspectable.try_as<ComboBoxDropDownEventRevokers>())
+            {
+                revokers->RevokeAll();
+            }
+        }
+        
+        bool shouldMonitorDropDownState = unbox_value<bool>(args.NewValue());
+        if (shouldMonitorDropDownState)
+        {
+            auto revokersInspectable = winrt::make<ComboBoxDropDownEventRevokers>();
+            auto revokers = revokersInspectable.try_as<ComboBoxDropDownEventRevokers>();
+
+            revokers->m_dropDownOpenedRevoker = comboBox.DropDownOpened(winrt::auto_revoke, ComboBoxHelper::OnDropDownOpened);
+            revokers->m_dropDownClosedRevoker = comboBox.DropDownClosed(winrt::auto_revoke, ComboBoxHelper::OnDropDownClosed);
+
+            comboBox.SetValue(DropDownEventRevokersProperty(), revokersInspectable);
+        }
+    }
+}
+
+void ComboBoxHelper::SetDropDownEventRevokers(winrt::UIElement const& target, winrt::IInspectable const& value)
+{
+    target.SetValue(s_DropDownEventRevokersProperty, ValueHelper<winrt::IInspectable>::BoxValueIfNecessary(value));
+}
+
+winrt::IInspectable ComboBoxHelper::GetDropDownEventRevokers(winrt::UIElement const& target)
+{
+    return ValueHelper<winrt::IInspectable>::CastOrUnbox(target.GetValue(s_DropDownEventRevokersProperty));
+}
+
+void ComboBoxHelper::OnDropDownOpened(const winrt::IInspectable& sender, const winrt::IInspectable& args)
+{
+    if (auto comboBox = sender.try_as<winrt::ComboBox>())
+    {
+        // We need to know whether the dropDown opens above or below the ComboBox in order to update corner radius correctly.
+        // Sometimes TransformToPoint value is incorrect because popup is not fully opened when this function gets called.
+        // Use dispatcher to make sure we get correct VerticalOffset.
+        DispatcherHelper dispatcherHelper;
+        dispatcherHelper.RunAsync([comboBox]()
+            {
+                UpdateCornerRadius(comboBox, /*IsDropDownOpen=*/true);
+            });
+    }
+}
+
+void ComboBoxHelper::OnDropDownClosed(const winrt::IInspectable& sender, const winrt::IInspectable& args)
+{
+    if (auto comboBox = sender.try_as<winrt::ComboBox>())
+    {
+        UpdateCornerRadius(comboBox, /*IsDropDownOpen=*/false);
+    }
+}
+
+void ComboBoxHelper::UpdateCornerRadius(const winrt::ComboBox& comboBox, bool isDropDownOpen)
+{
+    if (comboBox.IsEditable())
+    {
+        // TODO, read default values from theme resource once we have them
```

Comment:
Is there an Issue tracking this work? If so, reference it here.

Reply:
It's part of the corner radius work to add the resource keys for default values, so i should be able to get on this soon, I put it in a OneNote page to make sure I remember to come back and update it : )